src/main/java/org/openstreetmap/atlas/checks/validation/relations/OneMemberRelationCheck.java

```diff
@@ -0,0 +1,83 @@
+package org.openstreetmap.atlas.checks.validation.relations;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.tags.RelationTypeTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check flags {@link Relation}s which contain only one member.
+ *
+ * @author savannahostrowski
+ */
+public class OneMemberRelationCheck extends BaseCheck
+{
+    public static final String OMR_INSTRUCTIONS = "This relation, {0,number,#}, contains only "
+            + "one member.";
+
+    private static final List<String> FALLBACK_INSTRUCTIONS = Collections
+            .singletonList(OMR_INSTRUCTIONS);
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public OneMemberRelationCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Relation && !this.isFlagged(object.getOsmIdentifier());
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Relation relation = (Relation) object;
+        // Initialize a flag that will be flipped if the relation is found to be problematic
+        boolean flag = false;
+
+        // If the relation is a multipolygon
+        if (relation.isMultiPolygon())
+        {
+            // Determine if there are members with the role:inner
+            final boolean isInner = relation.members().stream().anyMatch(member -> member.getRole()
+                    .equalsIgnoreCase(RelationTypeTag.MULTIPOLYGON_ROLE_INNER));
+
+            // If the only member of the relation has the role:inner then we want to flag the
+            // relation
+            if (isInner && relation.members().size() == 1)
```

Comment:
For every multi-polygon relation this method will be looking for a member with `role:inner`. However, that should be only needed when the number of members is 1. I'd suggest updating logic here to avoid doing redundant work. Also `.members()` call is returning a new `Set` every time it is being called. I'd suggest caching it.

A conditional like `members.size() == 1 && (!relation.isMultiPolygon() || members.iterator().next().getRole().equalsIgnoreCase(RelationTypeTag.MULTIPOLYGON_ROLE_INNER))` should be able to filter relations that needs to be flagged.

Reply:
Wow, love it. So slick. Thanks @mertemin.